### PR TITLE
[calculations] Add findObservationResponseHoles

### PR DESCRIPTION
**Pull request by n-h-diaz**
Step 1 of computing calculations: This just finds the holes from the base response which will be candidates for calculation

### CODE REVIEW

File: internal/server/v2/observation/calculation_util.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// limitations under the License.
+
+package observation
+
+import (
+	pbv2 "github.com/datacommonsorg/mixer/internal/proto/v2"
+)
+
+// Given an input ObservationResponse, generate a map of variable -> entities with missing data.
+func findObservationResponseHoles(inputReq *pbv2.ObservationRequest, inputResp *pbv2.ObservationResponse) map[string]*pbv2.DcidOrExpression {
```
Comment (shifucun) [inline internal/server/v2/observation/calculation_util.go:22]: nit, i would usually format the func with long args as:

func funName(
  arg1 ...,
  arg2 ...,
) retType {
}

File: internal/server/v2/observation/calculation_util.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	result := map[string]*pbv2.DcidOrExpression{}
+	if inputReq.Variable.GetFormula() != "" {
+		// Currently do not support nested formulas.
```
Comment (shifucun) [inline internal/server/v2/observation/calculation_util.go:25]: shall we raise exception here to make this more explicit?

File: internal/server/v2/observation/calculation_util.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		return result
+	}
+	for variable, variableObs := range inputResp.ByVariable {
+		if len(inputReq.Entity.GetDcids()) > 0 {
+			emptyDcids := []string{}
```
Comment (shifucun) [inline internal/server/v2/observation/calculation_util.go:30]: nit: holeEntities (to match function name and intent)

**Review by shifucun — COMMENTED** [review]
nice PR! This is a really good start

File: internal/server/v2/observation/calculation_util.go (hunk 1 above)
Comment (n-h-diaz) [reply internal/server/v2/observation/calculation_util.go]: Done

File: internal/server/v2/observation/calculation_util.go (hunk 2 above)
Comment (n-h-diaz) [reply internal/server/v2/observation/calculation_util.go]: Done

File: internal/server/v2/observation/calculation_util.go (hunk 3 above)
Comment (n-h-diaz) [reply internal/server/v2/observation/calculation_util.go]: Done

**Pull request merged**